client/my-sites/plans-features-main/index.tsx

```diff
@@ -589,6 +591,12 @@ const PlansFeaturesMain = ( {
 					status: 'enabled',
 				},
 			};
+
+			if ( ! isEligibleForTrial && intentFromProps === 'plans-new-hosted-site' ) {
```

Comment:
should this be conditioning on `intent` instead? 🤔

Reply:
I don't have a preference. Used the same [we're using here](https://github.com/Automattic/wp-calypso/blob/add/not-eligible-message-to-plans-selection/client/my-sites/plans-features-main/index.tsx#L602). Should we pass that `intent` instead `intentFromProps` and use it on both?

Reply:
I believe the only time that `intent` may change is when we click on the "escape hatch" to show all plans, in which case the `intent` changes to the `plans-default-wpcom` one. It might have been the case that for P2 we still wanted to pass these overrides irrespective of the `intent` changing.

Not sure what the requirement is here. If you see the "view all plans" action button and click it, then it will show all the WPCOM plans as you see on `/start/plans`. Do you still need this functionality in that case? If not, then `intent` ought to be preferred.

Sorry if this is confusing. It could (will) use more clarifications for sure 😬

Reply:
I will address it as a follow-up, as we discussed, thanks!